testcases/OpTestXive.py

```diff
@@ -0,0 +1,209 @@
+            self.cv_HMC.vterm_run_command(self.console, "grubby --args=xive=off --update-kernel=/boot/vmlinuz-`uname -r`")
+            time.sleep(5)
+            self.cv_HMC.vterm_run_command(self.console, "sync; sync;")
+            time.sleep(5)
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.restart_lpar()
+            self.cv_HMC.wait_login_prompt(self.console, username="root", password="passw0rd")
+            self.console = self.cv_HMC.get_console()
+            self.cv_HMC.vterm_run_command(self.console, "which stty && stty cols 300; which stty && stty rows 30;")
+            self.cv_HMC.vterm_run_command(self.console,"cat /proc/interrupts | grep -Eai 'xive'")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --remove-args=xive* --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "sync;sleep 10")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --args=xive=on --update-kernel=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.vterm_run_command(self.console, "sleep 5; sync; sleep 5;")
+            self.cv_HMC.vterm_run_command(self.console, "grubby --info=/boot/vmlinuz-`uname -r`")
+            self.cv_HMC.restart_lpar()
+            self.cv_HMC.wait_login_prompt(self.console, username="root", password="passw0rd")
+            self.console = self.cv_HMC.get_console()
+            self.cv_HMC.vterm_run_command(self.console,"cat /proc/interrupts | grep -Eai 'xive'")
+
+    def xive_cpu_check(self): 
+        '''
+        This function will validate interrupts handled by CPU's.
+        Make sure while running this function xive should be on
+        '''
+        self.console = self.cv_HMC.get_console()
+        self.cv_HMC.vterm_run_command(self.console, "which stty && stty cols 300; which stty && stty rows 30;")
+        res = self.cv_HMC.vterm_run_command(self.console, "ip route list | grep default | awk '{print $5}'")
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts") 
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        time.sleep(2)
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        self.cv_HMC.vterm_run_command(self.console, "ppc64_cpu --cores-on")
+        self.cv_HMC.vterm_run_command(self.console, "ppc64_cpu --cores-on=1")
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        time.sleep(2)
+        self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s" % res[0])
+        self.cv_HMC.vterm_run_command(self.console, "ppc64_cpu --smt=off")
+        tmp1 = self.cv_HMC.vterm_run_command(self.console, "cat /proc/interrupts | grep %s | awk '{print $2}'" % res[0])
+        time.sleep(2)
```

Comment:
why do we need this ?

Reply:
Here am checking CPU0 count has increased or not after smt=0ff.So am giving 2sec of time to check that